Documentation/Rust Style.md

```diff
@@ -0,0 +1,36 @@
+# Contributing Rust code to MaidSafe
+
+We don't maintain a separate style guide but in general try to follow [common good practice](https://github.com/rust-lang/rust/tree/master/src/doc/style), write readable and idiomatic code and aim for full test coverage. In addition, this document lists a few decisions we've reached in discussions about specific topics.
+
+
+## Unwrap
+
+Don't unwrap [`Option`](https://doc.rust-lang.org/std/option/enum.Option.html)s or [`Result`](https://doc.rust-lang.org/std/result/enum.Result.html)s, except possibly when:
+
+1. locking a mutex,
+2. spawning a thread,
+3. joining a thread
+
+or in other patterns where using them makes the code _much simpler_ and it is _obvious at first glance_ to the reader (even one unfamiliar with the code) that the value cannot be `None`/`Err`.
+
+In these cases, as well as in tests, consider using the macros from the [`unwrap` crate](https://crates.io/crates/unwrap).
+
+
+## Rustfmt
+
+Apply `rustfmt` to new code before committing.
```

Comment:
That is automatically picked up by `rustfmt` if it is there, isn't it?
So maybe we shouldn't even mention it explicitly in the document: If we gradually add it to all crates, it will be automatic.